druid/src/widget/style_child.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A widget that themes its child.
+
+use crate::{
+    BaseState, BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, PaintCtx, Size, UpdateCtx,
+    Widget,
+};
+
+/// A widget that accepts a closure to update the theme for its child.
+pub struct StyleChild<T: Data, F: Fn(&mut Env) + 'static> {
+    f: F,
+    child: Box<dyn Widget<T>>,
```

Comment:
The more I think about it the more I think we want to not box single children automatically, and instead be generic over the type of the widget; see `Scroll` for an example of what I mean.

This won't be useful right now, but I have an idea for how we might let widgets reach down into the tree through multiple children; I'll try and write that up soon.

Reply:
Looking forward to seeing that. Given that the parent owns the subtree outright, I think it's architecturally valid to "reach in", but the details about how to do that cleanly are not clear to me.

Reply:
I copied the Scroll example's setup, including using WidgetPod. Does it look correct now?

Reply:
I think we maybe _don't_ need widgetpod? the rationale being that we're totally transparent (we don't modify behaviour at all), so we should be able to just pass things to the child directly?